### PR TITLE
BukkitSchedulerMock.java add missing methods

### PR DESCRIPTION
**Pull request by NeumimTo**
# Description
Implements isCurrentlyRunning and runTaskAsynchronously

# Checklist
The following items should be checked before the pull request can be merged.
- [ x] Code follows existing style.
- [ x] Unit tests added (if applicable).

### CODE REVIEW

**Comment by NeumimTo** [comment]
Locally all tests pass, so im just going to blame your ci and ignore the report, since i did not even touch code relevant to that failing test.

File: src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java [hunk 1]
```diff
@@ -527,8 +526,8 @@ public void runTask(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task)
 	@Override
 	public void runTaskAsynchronously(@NotNull Plugin plugin, @NotNull Consumer<BukkitTask> task)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		ScheduledTask scheduledTask = new ScheduledTask(id.getAndIncrement(), plugin, false, currentTick, task);
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java]:
```suggestion
		ScheduledTask scheduledTask = new ScheduledTask(this.id.getAndIncrement(), plugin, false, this.currentTick, task);
```

**Review by thelooter — CHANGES_REQUESTED** [review]
One small change

**Review by thelooter — APPROVED** [review]

**Comment by thelooter** [comment]
Not gonna merge just yet, since our CI isn't working rn. Will merge once that is fixed

**Pull request merged**